packages/cyberstorm/src/components/TextInput.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { MouseEventHandler, PropsWithChildren, ReactNode } from "react";
+import styles from "./componentStyles/TextInput.module.css";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faCircleX } from "@fortawesome/pro-regular-svg-icons";
+import { faMagnifyingGlass } from "@fortawesome/free-solid-svg-icons";
+
+export interface TextInputProps {
+  onChange?(event: React.ChangeEvent<HTMLInputElement>): void;
+  leftIcon?: ReactNode;
+  onLeftIconClick?: MouseEventHandler<HTMLButtonElement>;
+  rightIcon?: ReactNode;
+  onRightIconClick?: MouseEventHandler<HTMLButtonElement>;
+  onClear?: MouseEventHandler<HTMLButtonElement>;
+  placeholder?: string;
+  value?: string;
+  disabled?: boolean;
+  textInputStyle?: "root__small";
+}
+
+export const TextInput: React.FC<PropsWithChildren<TextInputProps>> = ({
+  onChange,
+  leftIcon = <FontAwesomeIcon fixedWidth icon={faMagnifyingGlass} />,
+  onLeftIconClick,
+  rightIcon,
+  onRightIconClick,
+  onClear,
+  placeholder = "Search",
+  value,
+  disabled = false,
+  textInputStyle,
+}) => {
+  return (
+    /* TS is not aware of defaultProps of function components. */
+    /* eslint-disable-next-line @typescript-eslint/no-non-null-assertion */
+    <div className={`${styles.root} ${styles[textInputStyle!]}`}>
+      <button
+        type="submit"
```

Comment:
Why submit?

Reply:
No reason, oversight, changed it to button like the others are.